expense/src/server/index.ts

```diff
@@ -26,42 +30,54 @@ async function run() {
   });
 
   app.post('/create', function(req, res) {
-    allExpenses[req.body.id] = expenseState.created;
+    allExpenses.set(req.body.id, expenseState.created);
 
     return res.json({ ok: 1 });
   });
 
   app.post('/action', function(req, res) {
-    const oldState = allExpenses[req.body.id];
+    const id = req.body.id;
+    const oldState = allExpenses.get(id);
     if (oldState == null) {
       throw new Error(`Invalid id ${req.body.id}`);
     }
+    let newState = '';
+
     switch (req.body.action) {
     case 'approve':
-      allExpenses[req.body.id] = expenseState.approved;
+      newState = expenseState.approved;
+      allExpenses.set(id, expenseState.approved);
       break;
     case 'reject':
-      allExpenses[req.body.id] = expenseState.rejected;
+      newState = expenseState.rejected;
+      allExpenses.set(id, expenseState.rejected);
       break;
     case 'payment':
-      allExpenses[req.body.id] = expenseState.completed;
+      newState = expenseState.completed;
+      allExpenses.set(id, expenseState.completed);
       break;
     default:
       throw new Error(`Invalid action ${req.body.action}`);
     }
 
+    if (oldState === expenseState.created &&
+        (newState === expenseState.approved || newState === expenseState.rejected)) {
+      // TODO: need to be able to complete an activity here like in Go
+      // https://github.com/temporalio/samples-go/blob/0248987774da74dceb265bb3f3b06e9d00e50a32/expense/server/main.go#L166
```

Comment:
Just use `Connection.service` to do this: https://nodejs.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.WorkflowService-1#respondactivitytaskcompleted

Reply:
We'll add a connection getter on activity context so you don't have to propagate the configuration to the activity.

Reply:
No longer necessary, removing this code.